lib/Github/Api/Search.php

```diff
@@ -61,6 +61,25 @@ public function code($q, $sort = 'updated', $order = 'desc')
         return $this->get('/search/code', ['q' => $q, 'sort' => $sort, 'order' => $order]);
     }
 
+    /**
+     * Search code by filter (q), but will return additional data to highlight
+     * the matched results.
+     *
+     * @link https://docs.github.com/en/rest/reference/search#text-match-metadata
+     *
+     * @param string $q     the filter
+     * @param string $sort  the sort field
+     * @param string $order asc/desc
+     *
+     * @return array list of code found
+     */
+    public function code_with_match($q, $sort = 'updated', $order = 'desc')
```

Comment:
```suggestion
    public function codeWithMatch(string $q, string $sort = 'updated', string $order = 'desc')
```

Also remove the redundant phpdoc after this change
```
     * @param string $q     the filter
     * @param string $sort  the sort field
     * @param string $order asc/desc
```